src/components/Config/Config.tsx

```diff
@@ -77,8 +78,20 @@ function Config() {
     }
   }, [network]);
 
+  const handleAlreadyConnected = useCallback(async () => {
+    checkMetamaskInstalled();
+    const connection = await connect();
```

Comment:
You know the desired network already so let's just pass it in to `connect`. You can see an example in `handleConfirmConfigured`. When you do it like this then the validation in `useAutomateConnection` should be enough to guide the user (meaning we don't need the new modal anymore), but please also change the error message for `connectedWrongNetwork` to `MetaMask is connected to Automate [...], please change the MetaMask network to Automate's [...].`